griptape/tools/google_docs/tool.py

```diff
@@ -0,0 +1,160 @@
+
+    def create_google_doc(self, file_name: str) -> InfoArtifact | ErrorArtifact:
+        try:
+            service = self._build_client(self.DOCS_SCOPES, "v1", "docs")
+            doc = service.documents().create(body={"title": file_name}).execute()
+            return InfoArtifact(f"Google Doc '{file_name}' created with ID: {doc['documentId']}")
+
+        except Exception as e:
+            logging.error(e)
+            return ErrorArtifact(f"Error creating Google Doc: {e}")
+
+    def append_text(self, params: dict, text: str) -> InfoArtifact | ErrorArtifact:
+        values = params["values"]
+        try:
+            service = self._build_client(self.DOCS_SCOPES, "v1", "docs")
+            doc = service.documents().get(documentId=values["document_id"]).execute()
+            content = doc['body']['content']
+            last_text = content[-1]['paragraph']['elements'][-1]['textRun']['content']
+            append_index = content[-1]['endIndex']
+            if last_text.endswith("\n"):
+                append_index -= 1
+
+            print("Appending at index:", append_index)
+
+            requests = [
+                {'insertText': {
+                    'location': {'index': append_index},
+                    'text': text
+                }}
+            ]
+
+            doc = service.documents().batchUpdate(documentId=values["document_id"],
+                                                  body={'requests': requests}).execute()
+            return InfoArtifact("Text appended successfully")
+
+        except Exception as e:
+            logging.error(e)
+            return ErrorArtifact(f"Error appending text to Google Doc: {e}")
+
+    def prepend_text(self, params: dict, text: str) -> InfoArtifact | ErrorArtifact:
+        values = params["values"]
+        try:
+            service = self._build_client(self.DOCS_SCOPES, "v1", "docs")
+            doc = service.documents().get(documentId=values["document_id"]).execute()
+            if len(doc['body']['content']) == 1:
+                requests = [
+                    {'insertText': {
+                        'location': {'index': 1},
+                        'text': text
+                    }}
+                ]
+            else:
+                start_index = doc['body']['content'][1]['startIndex']
+                requests = [
+                    {'insertText': {
+                        'location': {'index': start_index},
+                        'text': text
+                    }}
+                ]
+
+            doc = service.documents().batchUpdate(documentId=values["document_id"],
+                                                  body={'requests': requests}).execute()
+            return InfoArtifact("Text prepended successfully")
+
+        except Exception as e:
+            logging.error(e)
+            return ErrorArtifact(f"Error prepending text to Google Doc: {e}")
+
+    def upload_document(self, params: dict) -> InfoArtifact | ErrorArtifact:
```

Comment:
Missing `@activity` decorator.

Reply:
Removed function for now, will implement correctly